airflow/executors/kubernetes_executor.py

```diff
@@ -45,199 +47,6 @@
 MAX_LABEL_LEN = 63
 
 
-class KubeConfig:  # pylint: disable=too-many-instance-attributes
```

Comment:
Where there any changes from this class when it was moved out of here?

And is there a reason for making this change? Leaving it where it was would _massively_ make it easier to review this change, and there is nothing wrong with related classes living in the same pyhon package

Reply:
The issue with `KubeConfig` existing in the `kubernetes_executor` module is that `WorkerConfiguration` takes `KubeConfig` as an argument. In order to get better linting, [I annotated that argument](https://github.com/apache/airflow/blob/9d0d3e3039aa37766003b639b11605a5675e4746/airflow/kubernetes/worker_configuration.py#L45), however this creates a circular dependency. `kubernetes_executor` module requires `WorkerConfiguration` and `WorkerConfiguration` requires `kubernetes_executor`. 

This refactor is also being done in #5256, but I feel as thought that PR had been a bit forgotten, and the conflicts might be a bit tedious at this point. I agree this change is unrelated and creates a lot of noise. I could quickly move this into another PR.

Reply:
@davlum Could you move the Kubeconfig back and just make that a seperate PR? I'll get that merged quickly once this one goes through. I think with front-facing changes it better to put refactors in separately.